java/ql/src/experimental/Security/CWE/CWE-918/RequestForgery.qll

```diff
@@ -167,6 +168,61 @@ private class RequestEntityUriArg extends RequestForgerySink {
   }
 }
 
+/**
+ * A class representing a sink for remote class loading mechanism.
+ */
+private class URLClassloaderSink extends RequestForgerySink {
+  URLClassloaderSink() {
+
+    exists(MethodAccess e |
+      e.getMethod().getDeclaringType().getASupertype*() instanceof TypeClassLoader and
+      e.getQualifier() = this.asExpr() and
+      e.getMethod().getName() in ["loadClass", "findResource", "findResources", "getResourceAsStream"]
+    ) 
```

Comment:
Shouldnt this be restricted to `URLClassLoader`?

Reply:
Also note that loading a class from a user-controllable URLclassloader will result in RCE and probably should be reported as that to make severity more accurate

Reply:
> Shouldnt this be restricted to `URLClassLoader`?

"findResource", "findResources", "getResourceAsStream" these methods are from URLClassLoader, but "loadClass" comes from ClassLoader class

Reply:
We should check the declaring type, but also the actual instance type

Reply:
> Also note that loading a class from a user-controllable URLclassloader will result in RCE and probably should be reported as that to make severity more accurate

Do you mean to mention the RCE case here: https://github.com/github/securitylab/issues/312?

Reply:
Let me discuss it internally, but I meant to move it to a different query within the same submission

Reply:
We decided to leave it as-is for now, and potentially promote it to a different query in the future if it fits better.

Reply:
Ok thanks :)